src/plone/meta/pep_420.py

```diff
@@ -0,0 +1,186 @@
+        action="store_false",
+        default=True,
+        help='Don\'t "git commit" changes made by this script.',
+    )
+    parser.add_argument(
+        "--push",
+        dest="push",
+        action="store_true",
+        default=False,
+        help="Push changes directly.",
+    )
+    parser.add_argument(
+        "--interactive",
+        dest="interactive",
+        action="store_true",
+        default=False,
+        help="Run interactively: Scripts will prompt for input. Implies "
+        "--no-commit, changes will not be committed and pushed automatically.",
+    )
+    parser.add_argument(
+        "--no-tests",
+        dest="run_tests",
+        action="store_false",
+        default=True,
+        help="Skip running unit tests.",
+    )
+
+    args = parser.parse_args()
+    path = args.path.absolute()
+
+    if not (path / ".git").exists():
+        raise ValueError("`path` does not point to a git clone of a repository!")
+    if not (path / ".meta.toml").exists():
+        raise ValueError("The repository `path` points to has no .meta.toml!")
+
+    with change_dir(path) as cwd_str:
+        cwd = pathlib.Path(cwd_str)
+        bin_dir = cwd / "venv" / "bin"
+        branch_name = args.branch_name or "pep-420-native-namespace"
+        updating = git_branch(branch_name)
+
+        non_interactive_params = []
+        if not args.interactive and args.commit:
+            non_interactive_params = ["--no-input"]
+        else:
+            args.commit = False
+
+        if args.breaking:
+            call(bin_dir / "bumpversion", "--breaking", *non_interactive_params)
+        (path / "news" / "3928.breaking").write_text(
+            "Replace ``pkg_resources`` namespace with PEP 420 native namespace.\n"
+            "Support only Plone 6.2 and Python 3.10+.\n"
+        )
+
+        setup_py = []
+        for line in (path / "setup.py").read_text().splitlines():
+            if "from setuptools import find_packages" in line:
+                continue
+            elif "namespace_packages" in line:
+                continue
+            elif "packages=" in line:
+                continue
+            elif "package_dir=" in line:
+                continue
+            elif "zope.testrunner" in line:
+                setup_py.append(
+                    line.replace("zope.testrunner", "zope.testrunner >= 6.4")
+                )
+            elif "Framework :: Plone :: 6.0" in line:
+                continue
+            elif "Framework :: Plone :: 6.1" in line:
+                continue
+            elif "Programming Language :: Python :: 3.8" in line:
+                continue
+            elif "Programming Language :: Python :: 3.9" in line:
+                continue
+            elif 'python_requires=">=3.8"' in line:
+                setup_py.append(
+                    line.replace('python_requires=">=3.8"', 'python_requires=">=3.10"')
+                )
+            elif 'python_requires=">=3.9"' in line:
```

Comment:
A quick scan, probably incomplete, showed that at least `plone.app.testing` still had `3.7` on it, fixed it on https://github.com/plone/plone.app.testing/pull/112